Splunk Pre Release - handle offset bug (#15615)
demisto/content: Packs/SplunkPyPreRelease/Integrations/SplunkPyPreRelease/SplunkPyPreRelease_description.md

To create an authentication token, go to [Splunk create authentication tokens](https://docs.splunk.com/Documentation/SplunkCloud/8.1.2101/Security/CreateAuthTokens).
***

# Splunk Enterprise Security Users

## Fetching notable events.
The integration allows for fetching Splunk notable events using a default query. The query can be changed and modified to support different Splunk use cases. (See [Existing users](#existing-users)).

> We are highly recommend you to read [Ingest Incidents from a SIEM Using Splunk article](https://docs.paloaltonetworks.com/cortex/cortex-xsoar/6-2/cortex-xsoar-tutorials/tutorials/splunk-use-case.html) before starting using this integration.

Reviewer: ```suggestion
We highly recommend reading the [Ingest Incidents from a SIEM Using Splunk article](https://docs.paloaltonetworks.com/cortex/cortex-xsoar/6-2/cortex-xsoar-tutorials/tutorials/splunk-use-case.html) before starting using this integration.
```

11. Keep the defaults for fetch limit, first fetch timestamp, earliest time to fetch and latest time to fetch.
12. To add mirroring in both environments, in the Incident Mirroring Direction field, select Incoming and Outgoing.
Outgoing miroriring is recommended for Cortex XSOAR version 6.2 and above. If you enable mirroring, you need to add the timezone of the splunk server (in minutes). For example, if using GMT and the timezone is GMT +3 hours, set the timezone to +180. For UTC, set the timezone to 0. Set this only if the Splunk server is different than the Cortex XSOAR server. This is relevant only for fetching notable events.
13. Select Close Mirrored XSOAR Incident and Close Mirrored Splunk Notable Event, so when closing in one environment, it closes in the other.
14. In the Enrichment Types field, select Asset, Drilldown and Identity.
This enrichment provides additional information about Assets, drilldown, and identities that are related to the notable events you ingest.

> 15. Fetch backwards window - this backward window is for cases there is a gap between the event occurrence time and the event index time on the server.
> The backward window should be the average time in minutes between events occurence time and index time. This value will be included on the search query time range in order to not miss events that was index on delay.  

Reviewer: Would love to hear your thoughts and feedback
```suggestion
15. Fetch backwards window - this backward window is for cases there is a gap between the event occurrence time and the event index time on the server.
In Splunk, there is often a delay between the time an incident is created (the event's "occurrence time") and the time it is actually searchable in Splunk and visible in the index (the event's "index time").
This delay can be caused by an inefficient Splunk architecture, causing higher event indexing latency. However, it can also be "by design", e.g. if some endpoints / machines that generate Splunk events are usually offline.
Another point to note is that Splunk's searches are based on the occurrence time behind the scenes. Meaning, Splunk itself uses occurrence time as its determining factor for bucket division and search.
Therefore - we can't use index time as our primary search key without making the searches inefficient.
The backwards window is a way for you to configure the longest delay you would like to support.
This parameter determines the size of the occurrence time "sliding window" we will support in our queries. For example, if set for 2 hours, we will always search for events that occurred up to 2 hours ago (and will of course ignore duplicates).
However, there is obviously a price - the larger the window, the longer it will take for fetch queries to complete.
The best value to set depends on the delays that you see in your system (consult with your Splunk expert / master), the number of events in your system and other network properties.
Please use this parameter with careful consideration.
```
